src/web5-connect/web5-connect-server.ts

```diff
@@ -102,10 +107,15 @@ export class Web5ConnectServer {
    * Gets the Web5 Connect Response object. The `state` string can only be used once.
    */
   public async getWeb5ConnectResponse(state: string): Promise<Web5ConnectResponse | undefined> {
-    const response = this. cache.get(`response:${state}`);
-
-    // Delete the Response object from the data store now that it has been retrieved.
-    this.cache.delete(`response:${state}`);
+    const response = await this.cache.get(`response:${state}`);
+
+    // Delete the Response object from the cache once it has been retrieved.
+    // IMPORTANT: only delete if object is fetched, otherwise there could be a race condition
+    // where the object is not fetched in this call but become available immediately right,
+    // we would end up deleting it before a successful fetch.
```

Comment:
nit: same comment as above

```suggestion
    // Delete the Response object from the cache once it has been retrieved.
    // IMPORTANT: only delete if the object exists, otherwise there could be a race condition
    // where the object does not exist in this call but becomes available immediately after,
    // we would end up deleting it before it is successfully retrieved.
```